### ISSUE TITLE
speaker-test: allow controlling how long the noise is heard by the user

### ISSUE THREAD

**Issue by jidanni** [alsa-project/alsa-utils#219]
$ speaker-test --nloops 1
makes a pink noise for about one second.

On the man page mention how to make this into say a half second,
as some users would like to listen to shorter test sounds,
and no matter if the user does
```
$ speaker-test --device hdmi:HDMI -c 2 --nloops 1 --period 5
$ speaker-test --device hdmi:HDMI -c 2 --nloops 1 --period 50
$ speaker-test --device hdmi:HDMI -c 2 --nloops 1 --period 500
$ speaker-test --device hdmi:HDMI -c 2 --nloops 1 --period 5000
```
what the user hears is exactly the same, therefore --period is not what
controls this item.

I am saying that some users can confirm their speakers work without
listing to a full second of pink noise, and thus would like a way to
listen to a shorter length.

As far as
```

       -b | --buffer TIME
              Use buffer size of TIME microseconds.  When 0 is given, use  the
              maximal buffer size.  The default value is 0.

       -p | --period TIME
              Use  period size of TIME microseconds.  When 0 is given, the pe‐
              riods given by -P option is used.  The default value is 0.
```
are concerned, they must be documenting something completely different.

Debian
Package: alsa-utils                      
Version: 1.2.8-1

**Comment by perexg** [comment]
Yep, it controls the audio buffering parameters. The duration for one loop is hardcoded to 3 seconds for the "all channels" test or 5 seconds for the single channel test (-s).

**Comment by dima4p** [comment]
I support the creation of the feature too. Option `-d` seems to be good.
